habitat-lab/habitat/core/env.py

```diff
@@ -128,6 +128,28 @@ def __init__(
             }
         )
         self.action_space = self._task.action_space
+
+        # Sharing controller between actions
+        # This should go somewhere else, but it is unclear where
+        # we want to share a controller across actions since the controller
+        # is stateful
+        from habitat_baselines.articulated_agent_controllers import (
+            HumanoidRearrangeController,
+        )
+
+        walk_pose_path = (
+            "data/humanoids/humanoid_data/walking_motion_processed.pkl"
+        )
+        humanoid_controller = HumanoidRearrangeController(walk_pose_path)
+        if (
```

Comment:
At a glance, `core/env.py` looks to be task-agnostic code. It seems hacky to have this task-specific logic in here.

Reply:
Addressed in the latest commit